packages/router/router.ts

```diff
@@ -900,7 +905,12 @@ export function createRouter(init: RouterInit): Router {
         location: createLocation(state.location, result.location),
         ...submission,
       };
-      await startRedirectNavigation(result, redirectNavigation);
+      // By default we use a push redirect here since the user redirecting from
+      // the action already handles avoiding us backing into the POST navigation
+      // However, if they specifically used <Form replace={true}> we should
+      // respect that
+      let isPush = opts?.replace !== true;
+      await startRedirectNavigation(result, redirectNavigation, isPush);
```

Comment:
We needed to proxy the original `opts.replace` flag through to here, so that we can detect whether the user explicitly asked for `replace`(in which case we respect it) or if we should override the replace with a PUSH due to the redirect